pymc_marketing/customer_choice/nested_logit.py

```diff
@@ -0,0 +1,1064 @@
+                        f"prod_{n}_t", (P_nest[:, pt.newaxis] * (P_y_given_nest))
+                    )
+                path_prods_t.append(prod)
+            P_ = pm.Deterministic(
+                "p", pm.math.concatenate(path_prods_t, axis=1), dims=("obs", "alts")
+            )
+
+            _ = pm.Categorical("likelihood", p=P_, observed=y_data, dims="obs")
+
+        self.model = model
+        return model
+
+    def _data_setter(
+        self,
+        X: np.ndarray | pd.DataFrame,
+        y: np.ndarray | pd.Series | None = None,
+    ) -> None:
+        """Set the data.
+
+        Required from the parent class
+
+        """
+
+    def create_idata_attrs(self) -> dict[str, str]:
+        """Create the attributes for the InferenceData object.
+
+        Returns
+        -------
+        dict[str, str]
+            The attributes for the InferenceData object.
+
+        """
+        attrs = super().create_idata_attrs()
+        attrs["covariates"] = json.dumps(self.covariates)
+        attrs["depvar"] = json.dumps(self.depvar)
+        attrs["choice_df"] = json.dumps("Placeholder for DF")
+        attrs["nesting_structure"] = json.dumps(self.nesting_structure)
+        attrs["utility_equations"] = json.dumps(self.utility_equations)
+
+        return attrs
+
+    def sample_prior_predictive(  # type: ignore[override]
+        self, extend_idata: bool, kwargs: dict[str, Any]
+    ) -> None:  # type: ignore[override]
+        """Sample Prior Predictive Distribution."""
+        with self.model:  # sample with new input data
+            prior_pred: az.InferenceData = pm.sample_prior_predictive(500, **kwargs)
+            self.set_idata_attrs(prior_pred)
+
+        if extend_idata:
+            if self.idata is not None:
+                self.idata.extend(prior_pred)
+            else:
+                self.idata = prior_pred
+
+    def fit(self, extend_idata: bool, kwargs: dict[str, Any]) -> None:  # type: ignore[override]
+        """Fit Nested Logit Model."""
+        if extend_idata:
+            with self.model:
+                if self.idata is not None:
+                    self.idata.extend(pm.sample(**kwargs))
+        else:
+            with self.model:
+                self.idata = pm.sample(**kwargs)
+
+    def sample_posterior_predictive(  # type: ignore[override]
+        self, extend_idata: bool, kwargs: dict[str, Any]
+    ) -> None:
+        """Sample Posterior Predictive Distribution."""
+        if extend_idata:
+            with self.model:
+                if self.idata is not None:
+                    self.idata.extend(
+                        pm.sample_posterior_predictive(
+                            self.idata, var_names=["likelihood", "p"], **kwargs
+                        )
+                    )
+        else:
+            with self.model:
+                self.post_pred = pm.sample_posterior_predictive(
+                    self.idata, var_names=["likelihood", "p"], **kwargs
+                )
```

Comment:
nit: Can we sample_posterior_predictive then conditionally extend or assign instead of duplicating much of the logic

```python
with self.model: 
    idata = pm.sample_posterior_predictive(...)

if extend_idata ...
```